File: tests/gen_android/test_structure/expected/dd-sdk-android/src/main/java/com/datadog/android/bridge/DataStructure.kt
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2016-Present Datadog, Inc.
 */

package com.datadog.android.bridge

/**
 * A data structure to test data transfer through the bridge
 * @param someBoolean A mandatory boolean property
 * @param someInt A mandatory int property
 * @param someFloat A mandatory float property
 * @param someString A mandatory string property
 * @param someList A mandatory list property
 * @param someMap A mandatory map property
 */
data class DataStructure(
    val someBoolean: Boolean,
    val someInt: Int,
    val someFloat: Float,
    val someString: String,
    val someList: List<Any?>,
    val someMap: Map<String, Any?>
)

File: tests/gen_ios/test_structure/expected/dd-sdk-ios/Sources/Datadog/Bridge/DataStructure.swift
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2016-Present Datadog, Inc.
 */

import Foundation

/**
 A data structure to test data transfer through the bridge
 - Parameters:
     - someBoolean: A mandatory boolean property
     - someInt: A mandatory int property
     - someFloat: A mandatory float property
     - someString: A mandatory string property
     - someList: A mandatory list property
     - someMap: A mandatory map property
 */
struct DataStructure{
    let someBoolean: Bool,
    let someInt: Int,
    let someFloat: Float,
    let someString: NSString,
    let someList: NSArray,
    let someMap: NSDictionary
}
